src/python/pants/backend/python/target_types_rules.py

```diff
@@ -141,23 +142,67 @@ class InjectPythonDistributionDependencies(InjectDependenciesRequest):
 
 @rule
 async def inject_python_distribution_dependencies(
-    request: InjectPythonDistributionDependencies,
+    request: InjectPythonDistributionDependencies, python_infer_subsystem: PythonInferSubsystem
 ) -> InjectedDependencies:
-    """Inject any `.with_binaries()` values, as it would be redundant to have to include in the
-    `dependencies` field."""
+    """Inject dependencies that we can infer from entry points in the distribution.
+
+    Inject module owners referred to from `entry_points`.
+
+    (`.with_binaries()` is deprecated) Inject any `.with_binaries()` values, as
+    it would be redundant to have to include in the `dependencies` field.
+    """
     original_tgt = await Get(WrappedTarget, Address, request.dependencies_field.address)
+
+    entry_points = original_tgt.target[PythonDistributionEntryPoints].value
```

Comment:
This will fail if the target does not have the field. Instead, you can use `target.get(PythonDistributionEntryPoints)`.